packages/core/src/external/aws/lambda-logic/document-uploader.ts

```diff
@@ -49,7 +57,10 @@ export async function documentUploaderHandler(
   }
 
   // Get file info from the copied file
-  const { size, contentType } = await s3Utils.getFileInfoFromS3(destinationKey, destinationBucket);
+  const { size, contentType, eTag } = await s3Utils.getFileInfoFromS3(
+    destinationKey,
+    destinationBucket
+  );
```

Comment:
eTag is md5 hash of document. At documents greater than 5gb this is not exactly true but good enough for now.